docs/learning/tutorials/develop-agent.md

```diff
@@ -2,43 +2,48 @@
 
 
 ### Build
 - Open the cloned repo in VS Code
 - Open integrated terminal
-- `node ./common/scripts/install-run-rush install`
-- Edit agent-app/query-agent/src/QueryAgentConfig.ts
-    > `imjs_agent_imodel_name` = The name of your iModel<br/>
-`imjs_agent_client_id` = client_id of the registered agent application<br/>
-`imjs_agent_client_secret` = The client secret of registered agent application<br/>
-
+- `npm install`
+- Create a `.env` file at the project root with the following:
+    ``` ps
+    ###############################################################################
+    # This file contains secrets - don't commit or share it!
+    ###############################################################################
 
-- `node ./common/scripts/install-run-rush build -t imodel-query-agent`
+    # Specify an iModel
+    CONTEXT_ID=
+    IMODEL_ID=
 
-### Run
-- `cd agent-app\query-agent`
-- `npm run start`
-- The agent will listen for changesets for 40 seconds
+    # OIDC configuration
+    # Don't forget to add <CLIENT_ID>@apps.imsoidc.bentley.com to your CONNECT project. This can be done in the iModel Registration Dashboard.
+    CLIENT_ID=
+    CLIENT_SECRET=
+    ```
 
-In order to see changesets being logged, you can use the [imodel-changeset-test-utility](https://github.com/imodeljs/imodeljs-samples/tree/master/tools/imodel-changeset-test-utility).
+    > The values for `CONTEXT_ID` and `CONTEXT_ID` can be obtained from the IDs column of the [iModel registration dashboard](../../../getting-started/registration-dashboard/?tab=1)
 
-In a second terminal (`Ctrl+Shift+5` in VS Code's integrated terminal)
-- `cd tools\imodel-changeset-test-utility`
-- Edit tools\imodel-changeset-test-utility\src\ChangesetGenerationConfig.ts
-    > `imjs_agent_imodel_name` = The name of your iModel<br/>
-`imjs_agent_client_id` = client_id of the registered agent application<br/>
-`imjs_agent_client_secret` = cThe client secret of registered agent application<br/>
+    > The values for `CLIENT_ID` and `CLIENT_SECRET` come from the Agent Application you registered during the Setup step
 
-- `npm install`
 - `npm run build`
-- While the query-agent is running and listening for changesets...
-- `npm run start`
 
-You should see the query-agent receiving events and logging them to \agent-app\query-agent\lib\output\changeSummaries.
+### Run
+- `npm start`
+- The agent will listen for changesets pushed to iModelHub
+- Use the [iTwinSynchronizer](https://www.bentley.com/en/products/product-line/digital-twins/itwin-synchronizer) to synchronize a change and exercise the agent.
+- For testing, it can often also be useful to skip the event listening and just run against a specific changeset. To do that, either run `npm start -- --latest` to use the latest changeset, or `npm start -- --changeset=<CHANGESETID>` to use any specific changeset
+
+### Next Steps
+- Read the README for some more info

```

Comment:
Can you add a link to the README here or repo?

Reply:
What do you think about switching it to something like this, "Read the REAMDE for more information and how to start adding custom behavior" and then remove the next bullet.

Reply:
I feel like those are separate suggestions

Reply:
Yea, sorry, I meant to add two different comments here.

Reply:
I meant, that reading the README and implementing custom functionality feel like two different bullet points

Reply:
Plus, keeps the "rule of three" :)

Reply:
Okay, that's fine with me. I was mostly suggesting the removal of the direct reference to the `MyAgent.run` method since it's not really explained in this article but is in the README.  However, I do agree splitting it into multiple bullets makes it read a bit better.

Reply:
I'll generalize that the agent can be tinkered with:

- Implement custom functionality in the agent to fit a business use case